Sources/DebugExtensions/DebugProxy+Export.swift

```diff
@@ -0,0 +1,233 @@
+@MainActor private func openSavePanel(message: String, callback: @escaping (URL) -> Void) {
+    // Use the system panel for picking the folder where to save the files.
+    let openPanel = NSOpenPanel()
+    openPanel.message = message
+    openPanel.prompt = "Select"
+    openPanel.canChooseFiles = false
+    openPanel.canChooseDirectories = true
+    openPanel.canCreateDirectories = true
+
+    openPanel.begin { response in
+        guard response == .OK else { return }
+        callback(openPanel.url!)
+    }
+}
+
+#endif // AppKit
+
+
+// MARK: - iOS
+
+#if canImport(UIKit)
+
+import UIKit
+
+public extension CIImage.DebugProxy {
+
+    /// Renders the image with the given settings and opens a share sheet for exporting the image.
+    /// - Parameters:
+    ///   - filePrefix: A prefix that is added to the image file. By default, the name of the app is used.
+    ///   - codec: The codec of the exported image. Uses uncompressed TIFF by default.
+    ///   - format: The image format. This influences bit depth (8-, 16-, or 32-bit) and pixel format (uint or float). Defaults to 8-bit uint.
+    ///             Note that not all formats are supported by all codecs. E.g., `jpeg` and `heif` only supports 8-bit formats.
+    ///   - quality: The quality of the exported image, i.e., the amount of compression. Only supported by `jpeg`, `heif`, and `heif10` codecs.
+    ///   - colorSpace: The color space of the exported image. By default, the Display P3 color space is used.
+    ///                 Note that it needs to match the chosen `format`, i.e., a single-channel format needs a grayscale color space.
+    func export(filePrefix: String? = nil, codec: ImageCodec = .tiff, format: CIFormat = .RGBAh, quality: Float = 1.0, colorSpace: CGColorSpace? = nil) {
+        let filePrefix = exportFilePrefix(filePrefix: filePrefix)
+        let fileURL = FileManager.default.temporaryDirectory.appendingPathComponent("\(filePrefix)_image").appendingPathExtension(codec.fileExtension)
+        self.write(to: fileURL, codec: codec, format: format, quality: quality, colorSpace: colorSpace)
+        Task {
+            await openShareSheet(for: [fileURL])
+        }
+    }
+
+}
+
+public extension CIImage.DebugProxy.RenderResult {
+
+    /// Opens a share sheet for exporting the rendering artifacts (the image as TIFF and various rendering graphs as PDFs).
+    /// On macOS, the system dialog for picking an export folder will be shown instead.
+    /// - Parameter filePrefix: A prefix that is added to the files. By default, the name of the app is used.
+    func export(filePrefix: String? = nil) {
+        let filePrefix = exportFilePrefix(filePrefix: filePrefix)
+        let shareItems = self.writeResults(to: FileManager.default.temporaryDirectory, with: filePrefix)
+        Task {
+            await openShareSheet(for: shareItems)
+        }
+    }
+
+}
+
+/// Opens a share sheet for exporting the given files from the application's main window.
```

Comment:
Looks like this only opens a share sheet on iOS 13 and a document picker on later versions. Isn't the method name misleading then?